### PR TITLE
(WIP) (FM-7923) - Implement Puppet Strings

### PR DESCRIPTION
**Pull request by david22swan**

### CODE REVIEW

**Comment by ekohl** [comment]
Have you used a script to convert from the puppet doc format to puppet-strings? There are various modules in voxpupuli I'd also like to convert.

**Comment by david22swan** [comment]
@ekohl Unfortunately no. Did it all manually.

**Comment by david22swan** [comment]
Screen Shot 2019-04-15 at 2 27 32 PM

**Comment by ekohl** [comment]
https://github.com/voxpupuli/puppet-lint-param-docs can be used to enforce documentation btw

File: REFERENCE.md [hunk 1]
```diff
@@ -0,0 +1,4514 @@
...
+KDC spoofing attacks.
+
+Default value: 'on'
+
+##### `krb_servicename`
+
+Data type: `Any`
+
+Specifies the service name that will be used by Apache for authentication. Corresponding
+key of this name must be st
```
Comment (clairecadman) [inline REFERENCE.md]: an > a. 
Omit "that should be", so just ..."is a distinct application served by Passenger."

Comment (clairecadman) [inline REFERENCE.md]: Does this mean: "Sets PassengerRuby, a Ruby interpreter you use for the application on the virtual host" or "Sets PassengerRuby, a Ruby interpreter that the application uses on the virtual host."

Comment (clairecadman) [inline REFERENCE.md]: Same as above, the way it's worded is not that clear.

Comment (clairecadman) [inline REFERENCE.md]: Missing period.

Comment (david22swan) [reply REFERENCE.md]: Done

Comment (david22swan) [reply REFERENCE.md]: Changed syntax to `Sets [PassengerRuby](https://www.phusionpassenger.com/library/config/apache/reference/#passengerruby), specifying the Ruby interpreter that will be used on this virtual host.`

Comment (david22swan) [reply REFERENCE.md]: Changed to
`Sets [PassengerPython](https://www.phusionpassenger.com/library/config/apache/reference/#passengerpython), 
#   specifying the Python interpreter that will be used on this virutal host.`

Comment (david22swan) [reply REFERENCE.md]: added

Comment (clairecadman) [inline REFERENCE.md]: Omit "will", just "wait*s*". E.g. "...amount of time the preloader waits before shutting down..."

Comment (clairecadman) [inline REFERENCE.md]: "...the maximum number of requests for the application to process"?

Comment (clairecadman) [inline REFERENCE.md]: Type: "ammount" and "allow".  "...maximum amount of requests that *are* allowed to queue whenever..."

Comment (clairecadman) [inline REFERENCE.md]: Omit will. "Passenger immediately sends..."

Comment (clairecadman) [inline REFERENCE.md]: Type: "ammount" and "allow". "...maximum amount of requests that are allowed to queue whenever..."

Comment (david22swan) [reply REFERENCE.md]: think this comment is on the wrong line

Comment (clairecadman) [inline REFERENCE.md]: Omit will. "Passenger immediately sends..."

Comment (clairecadman) [inline REFERENCE.md]: "turns" > "turn"

Comment (david22swan) [reply REFERENCE.md]: NVM

Comment (clairecadman) [inline REFERENCE.md]: "applicationprocesses" > "application processes"?

Comment (david22swan) [reply REFERENCE.md]: No, it's right the way it is. An application can have multiple processes which process the requests. This variable set's the amount of processes each separate process can process.

Comment (david22swan) [reply REFERENCE.md]: I know it sounds weird, that's just how it is

Comment (david22swan) [reply REFERENCE.md]: updated

Comment (david22swan) [reply REFERENCE.md]: updated

Comment (david22swan) [reply REFERENCE.md]: updated to "...specify the maximum amount of time that requests are allowed to stay in the queue..."
Assumed above was a typo

Comment (david22swan) [reply REFERENCE.md]: updated

Comment (david22swan) [reply REFERENCE.md]: updated

Comment (david22swan) [reply REFERENCE.md]: updated

**Comment by tphoney** [comment]
- [Classes: apache::mod::\*](#classes-apachemodname) @david22swan  you have marked all the mod/* classes as private. They are not private, and should not be marked as such. I am aware there is a large amount to document, bit something would be better than not having any of them documented.

**Comment by eimlav** [comment]
Closing in favour of #1916 as @david22swan is no longer actively working on this module.

**Pull request closed without merging**